### PR TITLE
Add missing SecurityGroup selectors to EFS MountTarget

### PR DESCRIPTION
**Pull request by ytsarev**
### Description of your changes

* MountTarget of official provider was missing the selectors for SecurityGroup
* It is regression assuming scenario of migration from the community provider-aws, as the references are there https://marketplace.upbound.io/providers/crossplane-contrib/provider-aws/v0.33.0/resources/efs.aws.crossplane.io/MountTarget/v1alpha1#doc:spec-forProvider-securityGroupsRefs

Signed-off-by: Yury Tsarev <yury@upbound.io>
I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

uptest with the comment below

### CODE REVIEW

**Comment by ytsarev** [comment]
/test-examples="examples/efs/mounttarget.yaml"

**Comment by ytsarev** [comment]
/test-examples="examples/efs/mounttarget.yaml,examples/filesystem.yaml"

**Comment by ytsarev** [comment]
/test-examples="examples/efs/mounttarget.yaml,examples/efs/filesystem.yaml"

File: examples/efs/mounttarget.yaml [hunk 1]
```diff
@@ -38,4 +41,17 @@ spec:
     vpcIdSelector:
       matchLabels:
         testing.upbound.io/example-name: efs
-    cidrBlock: 172.16.10.0/24
\ No newline at end of file
+    cidrBlock: 172.16.10.0/24
+---
+apiVersion: ec2.aws.upbound.io/v1beta1
+kind: SecurityGroup
+metadata:
```
Comment (ulucinar) [inline examples/efs/mounttarget.yaml:48]: Could you please add the `meta.upbound.io/example-id` annotation to both the dependent & dependency resource manifests?

**Review by ulucinar — APPROVED** [review]
Thank you @ytsarev, lgtm.

File: examples/efs/mounttarget.yaml (hunk 1 above)
Comment (ytsarev) [reply examples/efs/mounttarget.yaml:48]: do we want to include just two resources or the whole dependency chain(down to VPC)?

**Comment by haarchri** [comment]
+1

File: examples/efs/mounttarget.yaml (hunk 1 above)
Comment (ytsarev) [reply examples/efs/mounttarget.yaml:54]: @ulucinar i've added the metadata to the whole example

**Comment by ytsarev** [comment]
/test-examples="examples/efs/mounttarget.yaml"

**Review by ulucinar — APPROVED** [review]
Thank you @ytsarev for adding all the dependencies, lgtm.

**Pull request merged**